astropy/config/paths.py

```diff
@@ -167,18 +168,22 @@ def _get_dir_path(cls, rootname: str) -> Path:
                 path.mkdir(exist_ok=True)
             return path.resolve()
 
-        if (
-            (dir_ := os.getenv(cls._directory_env_var)) is not None
-            and (xch := Path(dir_)).exists()
-            and not (xchpth := xch / rootname).is_symlink()
-        ):
-            if xchpth.exists():
-                return xchpth.resolve()
-
-            # symlink will be set to this if the directory is created
-            linkto = xchpth
-        else:
-            linkto = None
+        linkto: Path | None = None
+        env_dir = cls._xdg_env_dir
+        if (dir_ := os.getenv(env_dir)) is not None:
+            if (xch := Path(dir_)).is_dir():
+                if not (xchpth := xch / rootname).is_symlink():
+                    xchpth.mkdir(exist_ok=True)
+                    return xchpth.resolve()
+
+                # symlink will be set to this if the directory is created
+                linkto = xchpth
+            warn(
```

Comment:
Is there supposed to be else here?
```suggestion
        else:
            warn(
```

Reply:
not exactly, but it's true that there's still a scenario where the warning doesn't reflect the actual issue (the directory exists, but is a symlink), so I think I'll rewrite it as two different warnings. Thanks !

Reply:
Nevermind, I realized while trying to implement my previous answer that:
- you are absolutely right
- the deeply nested logic clearly makes this code hard to reason about

so I elected to restructure it using early returns instead of deep indentation levels. The final code include the correction you suggested. Let me know what you think. Thanks again for pointing it out !